src/org/traccar/api/BaseResource.java

```diff
@@ -29,4 +39,34 @@ protected long getUserId() {
         }
         return 0;
     }
+
+    protected void checkAndLinkPermission(LinkedHashMap<String, Long> entity, boolean link)
```

Comment:
I'm still not completely satisfied with the name. Maybe we should split it into check permissions?.. and then `linkObject` method is already there.